src/library/tlv.cpp

```diff
@@ -220,6 +220,71 @@ bool Tlv::IsValid() const
     {
         return false;
     }
+    else if (mScope == Scope::kNetworkDiag)
+    {
+        switch (mType)
+        {
+        // Network disgnostic layer TLVs
+        case Type::kNetworkDiagExtMacAddress:
+            return length == 8;
```

Comment:
We generally consider a TLV valid if its length is at least as long as the currently defined value length. This  accommodates potential future extensions of the TLV format.

Reply:
@abtink Thank you for your review. I have clarified the length of TLVs as per the definition in the spec. For instance, the length of the extended MAC address TLV in spec 5.19.2 is 8. Could you please kindly indicate any other inappropriate definitions in this PR?

Reply:
I think @abtink wants you to change the check to `return length >= 8;` so that the extended MAC extended address can be extended to maybe 16 bytes in Thread 2.0 or whatever later versions

Reply:
got it, Thanks @wgtdkp clarification. Let me update check condition.